sql/src/main/java/org/opensearch/sql/sql/SQLService.java

```diff
@@ -73,7 +73,8 @@ private AbstractPlan plan(
             "is not supported. Use `explain` for the initial query request.");
       }
       return queryExecutionFactory.create(request.getCursor().get(),
-        isExplainRequest, queryListener.orElse(null), explainListener.orElse(null));
+        isExplainRequest, request.isCursorCloseRequest(),
```

Comment:
[Echoing](https://github.com/opensearch-project/sql/pull/1660/files#r1206232124) Andrew's comments, this would look cleaner if `SQLService` analyzed a request and called appropriate method on `queryExecutionFactory` to create an `AbstractPlan`.